src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -145,6 +149,26 @@ public synchronized void launch(SlaveComputer computer, TaskListener listener) {
                     .orElse(null);
             node.setNamespace(namespace);
 
+            // register a namespace informer (if not registered yet) show relevant pod events in build logs
+            Map<String, SharedIndexInformer<Pod>> informers =
+                    node.getKubernetesCloud().getInformers();
+            if (informers.get(namespace) == null) {
+                Map<String, String> labelsFilter =
+                        new HashMap<>(node.getKubernetesCloud().getPodLabelsMap());
+                String jenkinsUrlLabel = sanitizeLabel(cloud.getJenkinsUrlOrNull());
+                if (jenkinsUrlLabel != null) {
+                    labelsFilter.put(PodTemplateBuilder.LABEL_KUBERNETES_CONTROLLER, jenkinsUrlLabel);
+                }
+                SharedIndexInformer<Pod> inform = client.pods()
+                        .inNamespace(namespace)
+                        .withLabels(labelsFilter)
+                        .inform(new PodStatusEventHandler(), TimeUnit.SECONDS.toMillis(30));
+                LOGGER.info(String.format(
+                        "Registered informer to watch pod events on namespace [%s], with labels [%s]",
+                        namespace, labelsFilter));
+                informers.put(namespace, inform);
+            }
```

Comment:
Use `Map#computeIfAbsent` for thread safety, probably simpler/more elegant to move the logic under `KubernetesCloud`.

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/1627/commits/66e538efbc67d76c450f5ae9b1b9c5e590378558 -> for thread safety